motionplan/planning_algorithms.go

```diff
@@ -17,17 +17,29 @@ const (
 	// default motion planner.
 	CBiRRT PlanningAlgorithm = "cbirrt"
 	// RRTStar indicates that an RRTStarConnectMotionPlanner should be used.
-	RRTStar = "rrtstar"
+	RRTStar PlanningAlgorithm = "rrtstar"
 	// TPSpace indicates that TPSpaceMotionPlanner should be used.
-	TPSpace = "tpspace"
+	TPSpace PlanningAlgorithm = "tpspace"
+	// UnspecifiedAlgorithm indicates that the use of our motion planning will accept whatever defaults the package
+	// provides. As of the creation of this comment, that algorithm will be cBiRRT.
+	UnspecifiedAlgorithm PlanningAlgorithm = ""
 )
 
+// AlgorithmSettings is a polymorphic representation of motion planning algorithms and their parameters. The `Algorithm`
+// should correlate with the available options (e.g. if `Algorithm` us CBiRRT, RRTStarOpts should be nil and CBirrtOpts should not).
+type AlgorithmSettings struct {
+	Algorithm   PlanningAlgorithm      `json:"algorithm"`
+	CBirrtOpts  *cbirrtOptions         `json:"cbirrt_settings"`
+	RRTStarOpts *rrtStarConnectOptions `json:"rrtstar_settings"`
+}
+
 type plannerConstructor func(
 	referenceframe.FrameSystem,
 	*rand.Rand,
 	logging.Logger,
 	*plannerOptions,
 	*ConstraintHandler,
+	*AlgorithmSettings,
```

Comment:
I am aware that you can get algorithm settings from `plannerOptions`, but I'm actually hoping to be able to remove the `*plannerOptions` argument in a follow-up PR

Reply:
Lets do this in the PR where you actually remove it